### PR TITLE
CouchDatabaseOperation and GetDocumentOperation

### PR DESCRIPTION
**Pull request by xpqz**

### CODE REVIEW

File: SwiftCloudant/Operations/GetDocumentOperation.swift [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    *
+    *  - document - The document read from the server
+    *
+    * - operationError - a pointer to an error object containing
+    *   information about an error executing the operation
+    */
+    var getDocumentCompletionBlock: (([String:AnyObject]?, ErrorType?) -> ())?
+
+    override func validate() -> Bool {
+        return true
```
Comment (rhyshort) [inline SwiftCloudant/Operations/GetDocumentOperation.swift]: This should validate that all the required fields are set, and call through to super.

**Comment by rhyshort** [comment]
So I found a way to fix the init problem and keep the type system happy, its a little hack and turns the session property into a computed property, so this needs a rebase to be able to compile :)

**Comment by rhyshort** [comment]
I went a head and merged, since I want to add support for the Swift package manager before I forget how.

**Comment by xpqz** [comment]
I've made a start on the Put - I pushed the branch even though not 100%
completed yet as I am on my home machine.

On 5 March 2016 at 00:23, Rhys Short <notifications@github.com> wrote:

> I went a head and merged, since I want to add support for the Swift
> package manager before I forget how.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rhyshort/swift-cloudant/pull/1#issuecomment-192534318>
> .
>



-- 
Stefan Kruger <stefan.kruger@gmail.com>

**Pull request merged**